src/convert.cpp

```diff
@@ -4,6 +4,20 @@
 #include <array>
 #include <cmath>
 #include <vector>
+#include <string>
+#include <sstream>
+
+namespace {
+
+template <typename T> std::string to_string(const T& value) {
+    std::ostringstream stream;
+    stream.setf(std::ios::fixed);
+    stream.precision(6);
+    stream << value;
+    return stream.str();
+}
```

Comment:
what's the reason for a function here, does Linux have issues with `std::to_string`?